src/main/java/org/opensearch/security/dlic/rest/api/AuthFailureListenersApiAction.java

```diff
@@ -0,0 +1,217 @@
+                builder.endObject();
+            }
+            builder.endObject();
+            return builder;
+        };
+    }
+
+    private void authFailureConfigApiRequestHandlers(RequestHandler.RequestHandlersBuilder requestHandlersBuilder) {
+
+        requestHandlersBuilder.override(
+            GET,
+            (channel, request, client) -> loadConfiguration(getConfigType(), false, false).valid(configuration -> {
+                final var config = (ConfigV7) configuration.getCEntry(CType.CONFIG.toLCString());
+                ok(channel, authFailureContent(config));
+            }).error((status, toXContent) -> response(channel, status, toXContent))
+        ).override(DELETE, (channel, request, client) -> loadConfiguration(getConfigType(), false, false).valid(configuration -> {
+            ConfigV7 config = (ConfigV7) configuration.getCEntry(CType.CONFIG.toLCString());
+
+            String listenerName = request.param("name");
+
+            // Try to remove the listener by name
+            if (config.dynamic.auth_failure_listeners.getListeners().remove(listenerName) == null) {
+                notFound(channel, "listener not found");
+            }
+            saveOrUpdateConfiguration(client, configuration, new OnSucessActionListener<>(channel) {
+                @Override
+                public void onResponse(IndexResponse indexResponse) {
+                    ok(channel, authFailureContent(config));
+                }
+            });
+        }).error((status, toXContent) -> response(channel, status, toXContent)))
+            .override(PUT, (channel, request, client) -> loadConfiguration(getConfigType(), false, false).valid(configuration -> {
+                ConfigV7 config = (ConfigV7) configuration.getCEntry(CType.CONFIG.toLCString());
+
+                String listenerName = request.param(NAME_JSON_PROPERTY);
+
+                ObjectNode body = (ObjectNode) DefaultObjectMapper.readTree(request.content().utf8ToString());
+                SecurityJsonNode authFailureListener = new SecurityJsonNode(body);
+                String authenticationBackend = authFailureListener.get(TYPE_JSON_PROPERTY).asString().equals(IP_TYPE)
+                    || authFailureListener.get(AUTHENTICATION_BACKEND_JSON_PROPERTY).isNull()
+                        ? null
+                        : authFailureListener.get(AUTHENTICATION_BACKEND_JSON_PROPERTY).asString();
+
+                // Try to remove the listener by name
+                config.dynamic.auth_failure_listeners.getListeners()
+                    .put(
+                        listenerName,
+                        new ConfigV7.AuthFailureListener(
+                            authFailureListener.get(TYPE_JSON_PROPERTY).asString(),
+                            Optional.ofNullable(authenticationBackend),
+                            Integer.parseInt(authFailureListener.get(ALLOWED_TRIES_JSON_PROPERTY).asString()),
```

Comment:
wdyt about adding `asInt()` to SecurityJsonNode and calling parseInt within there?

Reply:
Good idea: [fdaba0c](https://github.com/opensearch-project/security/pull/4641/commits/fdaba0cbf43cbb307e4ed762efd93e16ef3f5d4b)